src/vivarium/framework/results/manager.py

```diff
@@ -138,6 +141,8 @@ def gather_results(self, lifecycle_phase: str, event: Event) -> None:
         with 0.0.
         """
         population = self._prepare_population(event)
+        if population.empty:
+            return
```

Comment:
Why is this necessary?

Reply:
I guess it's not strictly necessary but seemed silly to pass an empty population through `gather_results`